libs/services/src/SlippageService/SlippageServiceMain.spec.ts

```diff
@@ -1,47 +1,445 @@
+      );
+
+      // WHEN: Get slippage
+      let result = await slippageService.getSlippageBps({
+        chainId,
+        baseTokenAddress,
+        quoteTokenAddress,
+      });
+
+      // THEN: We get the maximum slippage
+      expect(result).toBe(MAX_SLIPPAGE);
+
+      // WHEN: Get slippage (with the tokens inverted)
+      result = await slippageService.getSlippageBps({
+        chainId,
+        quoteTokenAddress,
+        baseTokenAddress,
+      });
+
+      // THEN: We get the maximum slippage too
+      expect(result).toBe(MAX_SLIPPAGE);
+    });
+
+    it(`one of the tokens has no prices available`, async () => {
+      getUsdPrices.mockImplementation(
+        async (chainId, tokenAddress, interval) => {
+          if (tokenAddress === baseTokenAddress) {
+            // GIVEN: One token is volatile
+            return POINTS_VOLATILITY_ZERO;
+          } else {
+            // GIVEN: The other token is not
+            return null;
+          }
+        }
+      );
+
+      // WHEN: Get slippage
+      let result = await slippageService.getSlippageBps({
+        chainId,
+        baseTokenAddress,
+        quoteTokenAddress,
+      });
+
+      // THEN: We get the maximum slippage
+      expect(result).toBe(MAX_SLIPPAGE);
+
+      // WHEN: Get slippage (with the tokens inverted)
+      result = await slippageService.getSlippageBps({
+        chainId,
+        quoteTokenAddress,
+        baseTokenAddress,
+      });
+
+      // THEN: We get the maximum slippage too
+      expect(result).toBe(MAX_SLIPPAGE);
+    });
+
+    it(`if the prices change a lot`, async () => {
+      // GIVEN: The prices don't change at all
+      getUsdPrices.mockResolvedValue(POINTS_WITH_HIGH_VOLATILITY);
+
+      // WHEN: Get slippage
+      const result = await slippageService.getSlippageBps({
+        chainId,
+        baseTokenAddress,
+        quoteTokenAddress,
+      });
+
+      // THEN: We get the maximum slippage
+      expect(result).toBe(MAX_SLIPPAGE);
+    });
+
+    it(`if the asset is volatile`, async () => {
+      // GIVEN: The prices have high volatility
+      getUsdPrices.mockResolvedValue(getPoints([100, 100.02, 100.04, 100.06]));
+
+      // GIVEN: Price is 1 USD
+      getUsdPrice.mockResolvedValue(1);
+
+      // WHEN: Get slippage
+      const result = await slippageService.getSlippageBps({
+        chainId,
+        baseTokenAddress,
+        quoteTokenAddress,
+      });
+
+      // THEN: We get the maximum slippage
+      //    AVG = (100 + 100.02 + 100.04 + 100.06)/4 = 100.03
+      //    VARIANCE = ((100 - 100.03)**2 + (100.02 - 100.03)**2 + (100.04 - 100.03)**2 + (100.06 - 100.03)**2) / 4 = 0.0005
+      //    STDDEV = sqrt(0.0005) = 0.02236067977
+      //    Volatility Fair Settlement (USD) = 0.02236067977 * sqrt(1) = 0.02236067977
+      //    Volatility Fair Settlement (Token) = 0.02236067977 / 1 = 0.02236067977
+      //    Slippage BPS = ceil(0.02236067977 * 10000) = 224
+      //    Adjusted Slippage = MAX = 200
```

Comment:
Thanks, this helps visualize it!

I still don't get line 177, same as I commented in the code.

Reply:
I hope is more clear now. I tried to explain in the thread above, and added more comments